src/tests/unit_tests/actions/test_reranking.py

```diff
@@ -0,0 +1,49 @@
+from unittest.mock import patch
+
+import pytest
+
+from sherpa_ai.actions.google_search import GoogleSearch
+from sherpa_ai.actions.reranking import RerankingByQuery
+
+
+Resources = [
+    {"Document": "0", "Source": "www.link1.com"},
+    {"Document": "1", "Source": "www.link2.com"},
+    {"Document": "2", "Source": "www.link3.com"},
+]
+
+
+@pytest.fixture
+def mock_google_search():
+    with patch("sherpa_ai.actions.GoogleSearch.search") as mock_google_search:
+        mock_google_search.return_value = Resources
+        yield mock_google_search
+
+
+@pytest.fixture
+def mock_embedding_func():
+    def embedding(text):
+        return_values = [[1, 0, 0], [1, 1, 1], [1, 1, 0]]
+        return return_values[int(text)]
+
+    return embedding
+
+
+def test_default_reranker(mock_google_search, mock_embedding_func):
+    reranker = RerankingByQuery(embedding_func=mock_embedding_func)
```

Comment:
we need to have a base embedding function class since some of them are not callable. For example, SentenceTransformerEmbeddings has embed_query instead of it self being callable.

Reply:
I'm hoping to have an adapter for all of them so that we will not directly use the embedding from the langchain. But I think create a base embedding is a good idea.